org.ekstep.reviewercomments-1.0/editor/plugin.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Plugin to get comments from portal
+ * @class reviewerComments
+ * @constructor
+ * @extends org.ekstep.contenteditor.basePlugin
+ * @author Revathi P 
+ * @listens stage:render:complete
+ */
+
+org.ekstep.contenteditor.basePlugin.extend({
+    NO_COMMENTS: 'No Reviewer Comments',
+    initialize: function () {
+        ecEditor.addEventListener("stage:render:complete", this.initializeComments, this);
+    },
+    initializeComments: function (event, callback) {
+        var instance = this;
+        if (instance.comments == undefined) {
+            this.context = org.ekstep.contenteditor.globalContext;
+            if (!ecEditor._.isUndefined(this.context)) {
+                if (!ecEditor._.isUndefined(this.context.contentId) && this.context.contentId != "") {
+                    var instance = this;
+                    var pkgVersion = ecEditor.getService('content').getContentMeta(org.ekstep.contenteditor.api.getContext('contentId')).pkgVersion;
+                    var data = {
+                        "request": {
+                            "contextDetails": {
+                                'contentId': ecEditor.getContext('contentId'),
+                                'contentType': 'application/vnd.ekstep.ecml-archive',
+                                'contentVer': !_.isUndefined(pkgVersion) ? pkgVersion.toString() : '0'
+                            }
+                        }
+                    };
+                    //Do Api call and get the response  
+                    ecEditor.getService(ServiceConstants.CONTENT_SERVICE).getComments(data, function (error, response) {
+                        if (!error) {
+                            //Loop through the response and generate the comments Thread
+                            // instance.comments = response.data.result.comments
+                            instance.showComments(response.data.result.comments);
+                        } else {
+                            instance.displayNoComments();
+                            ecEditor.dispatchEvent("org.ekstep.toaster:error", {
+                                message: "Error in fetching comments, please try again!",
+                                position: 'topCenter',
+                                icon: 'fa fa-warning'
+                            });
+                            ecEditor.getService('telemetry').error({
+                                "env": 'content',
+                                "stage": '',
```

Comment:
stage is required please get stage id and pass it

Reply:
If stageId is passed, that particular stageId's comment will be fetched. If stageId is empty, all the stage comments are fetched, the call is as per the design, https://project-sunbird.atlassian.net/wiki/pages/viewpage.action?pageId=743374958